### PR TITLE
Add Nix setup

### PR DESCRIPTION
**Pull request by alter2000**
I'm not sure how far out this is from being usable but the things that _should_ build are building and I'm not sure how to check whether things are working now.

**NOTE** that `parametrized` is [__not__ installed](https://github.com/channable/heliclockter/blob/ffbd0e94662dc60c4cb64cd7b98842433c3384c2/default.nix#L16) as it's not on [Nixpkgs unstable](https://search.nixos.org/packages?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=parametrized). I think I can add a definition but I'm not sure how many hiccups there would be.

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    pythonEnv = python.withPackages (p: [
+      p.bandit
+      p.black
+      p.mypy
+      p.pydantic
+      p.pylint
+      p.pytest
+      # TODO: actually add to overlay or nixpkgs
+      # p.parametrized
```
Comment (DanielNoord) [inline default.nix:16]: Quick review, but this doesn't support some of the newer Python versions anyway. Or at least, it is somewhat broken.

We should prefer `pytest.mark.parametrized`.

Comment (alter2000) [reply default.nix:16]: Noted, how would I go about adding it? I tried to google that string but the closest I got was `pytest.mark.parametrize` on the [docs](https://docs.pytest.org/en/6.2.x/parametrize.html) but I can't find any similar packages anywhere.

Comment (DanielNoord) [reply default.nix:16]: That's the one. It's almost a drop in replacement. Just need to change the test that use it

Comment (alter2000) [reply default.nix:16]: Okay then, I'll need to take a while to figure out more about it after lunch and meetings before I get comfortable with it.

Comment (alter2000) [reply default.nix:16]: Oh god it's param**e**terized that's why I couldn't find it :angery:

**Comment by alter2000** [comment]
I'm not sure whether this is still needed, in which case just reopen this and I can get it up to speed, but until then I'm leaving this up and fixed with `pytest.mark.parametrize` failing on `instantiation_test::test_future_and_past`:

```
ERROR collecting tests/instantiation_test.py
In test_future_and_past: function already takes an argument 'days' with a default value
```

**Pull request closed without merging**